Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.cpp

```diff
@@ -370,14 +377,18 @@ bool InPlaceAbstractState::merge(BasicBlock* from, BasicBlock* to)
             unsigned valueCountInFromBlock = 0;
             for (NodeAbstractValuePair& fromBlockValueAtTail : from->ssa->valuesAtTail) {
                 if (fromBlockValueAtTail.node == node) {
-                    ASSERT(fromBlockValueAtTail.value == forNode(node));
+                    if (node->op() == EnumeratorNextUpdateIndexAndMode)
```

Comment:
How about checking `node->isTuple()` instead since that was the fundamental reason of this custom code?